### PR TITLE
Fix expect for complex MPS

### LINKED ISSUE #866: [ITensors] [BUG] Error in expect due to missing check of roundoff imaginary parts
https://github.com/ITensor/ITensors.jl/blob/2ed0f58d287b0184247dbedad411b7583d92a08e/src/mps/mps.jl#L785

```julia
using ITensors

q = siteinds("Qubit", 5)
ψ = randomMPS(ComplexF64, q, 2)
expect(ψ, "X"; sites = 1)
```

Outputs:
```julia
ERROR: LoadError: InexactError: Float64(-0.21915457010994757 + 3.469446951953614e-18im)
Stacktrace:
 [1] Real
   @ ./complex.jl:44 [inlined]
 [2] convert(#unused#::Type{Float64}, x::ComplexF64)
   @ Base ./number.jl:7
 [3] expect(psi::MPS, ops::Tuple{String}; kwargs::Base.Pairs{Symbol, Int64, Tuple{Symbol}, NamedTuple{(:sites,), Tuple{Int64}}})
   @ ITensors ~/.julia/dev/ITensors/src/mps/mps.jl:785
 [4] #expect#877
   @ ~/.julia/dev/ITensors/src/mps/mps.jl:796 [inlined]
 [5] top-level scope
   @ ~/.julia/dev/ITensors/giacomo/issue_expect.jl:7
 [6] include(fname::String)
   @ Base.MainInclude ./client.jl:451
 [7] top-level scope
   @ REPL[13]:1
```

### PR DESCRIPTION
**Pull request by emstoudenmire**
Fixes #866, bug in `expect` for complex-valued inputs and Hermitian operators. Check if the element type is expected to be real and take the real part if so.

- Add timers to runtests.jl
- Test and fix #866
- Formatting

### CODE REVIEW

File: docs/make.jl [hunk 1]
```diff
@@ -2,4 +2,9 @@ include("settings.jl")
 
 makedocs(; sitename=sitename, settings...)
 
-deploydocs(; repo="github.com/ITensor/ITensors.jl.git", devbranch="main", push_preview=true, deploy_config=Documenter.GitHubActions())
```
Comment (emstoudenmire) [inline docs/make.jl:5]: This was just a leftover formatting thing unrelated to this PR.

File: test/mps.jl [hunk 2]
```diff
@@ -721,15 +721,15 @@ end
 
   @testset "expect Function" begin
     N = 8
-    s = siteinds("S=1/2", N; conserve_qns=false)
-    psi = randomMPS(s, n -> isodd(n) ? "Up" : "Dn"; linkdims=4)
```
Comment (emstoudenmire) [inline test/mps.jl:725]: Changed the test to trigger the error reported

File: test/runtests.jl [hunk 3]
```diff
@@ -54,14 +54,14 @@ if Threads.nthreads() == 1
       "examples.jl",
     ]
       println("Running $filename")
-      include(filename)
```
Comment (emstoudenmire) [inline test/runtests.jl:57]: Took advantage of this PR to add timers here

**Comment by mtfishman** [comment]
Thanks for the fix!

**Pull request merged**